src/openrct2/object/ObjectManager.cpp

```diff
@@ -589,14 +589,14 @@ class ObjectManager final : public IObjectManager
         std::mutex commonMutex;
         ParallelFor(requiredObjects, [this, &commonMutex, requiredObjects, &objects, &badObjects, &loadedObjects](size_t i) {
             auto ori = requiredObjects[i];
-            Object* loadedObject = nullptr;
+            std::unique_ptr<Object> object;
             if (ori != nullptr)
             {
-                loadedObject = ori->LoadedObject;
+                auto loadedObject = ori->LoadedObject;
```

Comment:
There is a difference here. It will never assign the already loaded object into objects[i] in case ```ori != nullptr```